syntax = "proto3";

package gateway_backend;

service Auth {
  rpc HasPermission (Permission) returns (IsPermitted) {}

  rpc HasRole (Role) returns (RoleGranted) {}

  rpc IsAuthorized (Token) returns (IsValid) {}
}

message Permission {
  string token = 1;
  string permission = 2;
}

message IsPermitted {
  bool has_permission = 1;
}

message Role {
  string token = 1;
  string role = 2;
}

message RoleGranted {
  bool has_role = 1;
}

message Token {
  string token = 1;
}

message IsValid {
  bool is_valid = 1;
}
